src/main/java/com/zendesk/maxwell/monitoring/MaxwellConfigServlet.java

```diff
@@ -0,0 +1,118 @@
+ * If property provided for update is null, it will be ignored.
+ */
+public class MaxwellConfigServlet extends HttpServlet {
+  private static final java.lang.String CONTENT_TYPE = "application/json";
+  private static final ObjectMapper mapper = new ObjectMapper();
+  static final Logger LOGGER = LoggerFactory.getLogger(MaxwellConfigServlet.class);
+  private final MaxwellContext context;
+
+
+  public MaxwellConfigServlet(MaxwellContext context) {
+    super();
+    this.context = context;
+  }
+
+  @JsonIgnoreProperties(ignoreUnknown = true)
+  private static class SerializedConfig {
+    public String filter;
+  }
+
+  private static class ErrorResponse {
+    public int code;
+    public String message;
+
+    public ErrorResponse(int code, String message) {
+      this.code = code;
+      this.message = message;
+    }
+
+    public static String getErrorResponseString(int code, String message) throws JsonProcessingException {
+      ErrorResponse errResponse = new ErrorResponse(code, message);
+      return String.format("{\"error\":%s}",  mapper.writeValueAsString(errResponse));
+    }
+  }
+
+  protected void doPatch(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    SerializedConfig sconfig;
+    try {
+      String reqbody = req.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+      sconfig = mapper.readValue(reqbody, SerializedConfig.class);
+    } catch(Exception e) {
+      resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      resp.setContentType(CONTENT_TYPE);
+      try (PrintWriter writer = resp.getWriter()) {
+        writer.println(ErrorResponse.getErrorResponseString(HttpServletResponse.SC_BAD_REQUEST, String.format("error processing body: %s", e.getMessage())));
+      }
+      return;
+    }
+
+    try {
+      applySerializedConfig(sconfig);
+
+      resp.setStatus(HttpServletResponse.SC_OK);
+      resp.setContentType(CONTENT_TYPE);
+      try (PrintWriter writer = resp.getWriter()) {
+        writer.println(getSerializedConfig());
+      }
+    } catch (InvalidFilterException e) {
+      resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      resp.setContentType(CONTENT_TYPE);
+      try (PrintWriter writer = resp.getWriter()) {
+        writer.println(ErrorResponse.getErrorResponseString(HttpServletResponse.SC_BAD_REQUEST, String.format("invalid filter: %s", e.getMessage())));
+      }
+    }
+  }
+
+  @Override protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    resp.setStatus(HttpServletResponse.SC_OK);
+    resp.setContentType(CONTENT_TYPE);
+      try (PrintWriter writer = resp.getWriter()) {
+        writer.println(getSerializedConfig());
+      }
+  }
+
+  @Override
+  public void service(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    if (request.getMethod().equalsIgnoreCase("PATCH")) {
```

Comment:
I think we should probably support PUT/POST too.  I don't mind if the semantics aren't precisely correct, I'd rather it follow principle of least surprise.